src/debian/11/helix/arm64v8/Dockerfile

```
@@ -43,20 +43,21 @@ RUN apt-get update \
 
 ENV LANG=en_US.utf8
 
-RUN ln -sf /usr/bin/python3 /usr/bin/python \
-    && pip download --no-deps helix-scripts --index-url https://dnceng.pkgs.visualstudio.com/public/_packaging/helix-client-prod/pypi/simple \
-    && export CRYPTOGRAPHY_DONT_BUILD_RUST=1 \
-    && pip install ./helix_scripts-*-py3-none-any.whl
+RUN ln -sf /usr/bin/python3 /usr/bin/python
 
-# Create helixbot users and give rights to sudo without password
-# (we use two users here to ensure volume mounting works with two possible UIDs of the host UID)
-# additionally, preinstall the virtualenv packages used for VSTS reporting to save time
-RUN /usr/sbin/adduser --disabled-password --gecos '' --uid 1001 --shell /bin/bash --ingroup adm helixbot \
-    && /usr/sbin/adduser --disabled-password --gecos '' --uid 1000 --shell /bin/bash --ingroup adm helixbot2 \
+# Create helixbot user and give rights to sudo without password
+RUN /usr/sbin/adduser --disabled-password --gecos '' --uid 1000 --shell /bin/bash --ingroup adm helixbot \
+    && usermod -aG sudo helixbot \
     && chmod 755 /root \
-    && echo "helixbot ALL=(ALL)       NOPASSWD: ALL" > /etc/sudoers \
-    && echo "helixbot2 ALL=(ALL)       NOPASSWD: ALL" > /etc/sudoers 
+    && echo "helixbot ALL=(ALL)       NOPASSWD: ALL" | (EDITOR="tee -a" visudo)
```

I just realized the reason `sudo` wasn't working in this image. problem is deleted line above replaced the whole /etc/sudoers file, removing `helixbot` user from it.

I could remove the `usermod` addition and go back to
```suggestion
    && echo "helixbot ALL=(ALL)       NOPASSWD: ALL" > /etc/sudoers
```
if you have strong feelings about this

I'm fine either way though `visudo` is strongly recommended everywhere anyone wants to edit /etc/sudoers and replacing that entire file is unnecessary

regardless of other opinions, I'll remove the `usermod` and see if that works locally

---

removing `usermod` worked

we have two choices from my perspective:
1. consistently use `> /etc/sudoers` and thereby replace the whole file
2. consistently use `| (EDITOR="tee -a" visudo)`

I lean toward the second option. it's somewhat further from what's in 'main' but follows accepted Linux practise. not sure of any real downside. I'll admit we could reliably use `>> /etc/sudoers` too but don't see the advantage over the `visudo` gesture

---

smallish reasons to prefer my second choice:
- this is one of the few ways the non-Helix containers are a bit more capable
- for example, `docker run ... --user root ...` and attempting to use `sudo` in the container results in `root is not in the sudoers file.  This incident will be reported.` in most Helix containers but not the revised Debian 11 ones

---

Honestly, I would prefer using `>> /etc/sudoers`. That is a more understood syntax by a broader set of people, IMO. I had no idea what `EDITOR="tee -a" visudo` did without looking it up. The benefits that `visudo` provides aren't necessary in this context.

---

hadn't noticed b/4 but recent Alpine images write to /etc/sudoers.d/helixbot instead of /etc/sudoers. the two files are essentially equivalent as long as we're appending to /etc/sudoers. I don't have a strong preference either way but suggest we use the same file in all Helix images. thoughts❓

---

Copilot says that writing to `/etc/sudoers.d/helixbot` is considered the better practice. Reference: https://unix.stackexchange.com/questions/375433/etc-sudoers-vs-etc-sudoers-d-file-for-enabling-sudo-for-a-user

---

done in latest iteration. I also rebased on latest main and squashed the commits. will update the PR description to match the latest in a minute

---

Why did you squash? That makes it very hard to discern the latest changes from everything else.

---

sorry. I did it b/c I wanted to ensure my description of the final changes was complete in the simplest way

literally the only changes from the previous iteration were the switch to the /etc/sudoers.d/helixbot